src/header.rs

```diff
@@ -547,6 +550,26 @@ impl NiftiHeader {
     }
 }
 
+/// Build the `dim` field of the `NiftiHeader` from a ndarray shape.
+///
+/// # Example
+///
+/// ```
+/// use ndarray::Array3;
+/// use nifti::header::build_dim_array;
+/// let im = Array3::<f32>::zeros((3, 4, 5));
+/// assert_eq!(build_dim_array(im.shape()), [3, 3, 4, 5, 1, 1, 1, 1]);
+/// ```
+#[cfg(feature = "ndarray_volumes")]
+pub fn build_dim_array(data_shape: &[usize]) -> [u16; 8] {
```

Comment:
If this is to be a public method, we should highlight the fact that it panics: add a `# Panics` section in the documentation and make it fail faster by checking the shape's length sooner.

```suggestion
pub fn build_dim_array(data_shape: &[usize]) -> [u16; 8] {
    if data_shape > 7 {
        panic!("cannot make NIfTI-1 shape of {} dimensions (maximum is 7)", data_shape.len());
    }
```